src/main.rs

```diff
@@ -150,13 +162,26 @@ fn main() -> Result<()> {
                 LAST.with(|last| *last.borrow_mut() = now);
                 let datetime: DateTime<Utc> = now.into();
                 let time = datetime.format("%H%M%S.00");
-                let gpgga = format!(
-                    "$GPGGA,{},{:02}{:010.7},{},{:03}{:010.7},{},4,12,1.3,{:.2},M,0.0,M,1.7,0078",
-                    time, lat_deg, lat_min, lat_dir, lon_deg, lon_min, lon_dir, heightf
-                );
-                let checksum = checksum(gpgga.as_bytes());
-                let gpgga = format!("{}*{:X}\r\n", gpgga, checksum);
-                buf.write_all(gpgga.as_bytes()).unwrap();
+                let message = match &opt.area_id {
+                    Some(area_id) => {
+                        let request_counter = match &opt.request_counter {
+                            Some(rc) => rc.to_string(),
+                            None => String::new()
+                        };
+                        let solution_id = match &opt.solution_id {
+                            Some(solution_id) => solution_id.to_string(),
+                            None => String::new()
+                        };
+                        format!("$PSWTCRA,{},{},,{},", request_counter, area_id, solution_id)
+                    },
+                    None => {
+                        format!("$GPGGA,{},{:02}{:010.7},{},{:03}{:010.7},{},4,12,1.3,{:.2},M,0.0,M,1.7,0078",
+                        time, lat_deg, lat_min, lat_dir, lon_deg, lon_min, lon_dir, heightf)
+                    }
+                };
+                let checksum = checksum(message.as_bytes());
+                let message = format!("{}*{:X}\r\n", message, checksum);
```

Comment:
I don't think we should add a `\r\n` if it's not there for GGA

Reply:
Ok

Reply:
My mistake. This was always there.
```
let checksum = checksum(gpgga.as_bytes());
let gpgga = format!("{}*{:X}\r\n", gpgga, checksum);
buf.write_all(gpgga.as_bytes()).unwrap();
```

Reply:
"this" being the carriage return